src/scripts/makeitsafe.js

```diff
@@ -0,0 +1,49 @@
+// TODO: Clean up. Not really wriotten for modules and a bit of a hack as is.
+// Imports should be better, also may as well make this a proper javascript
+// class opposed to whatever this is.
+
+var infiniteLoopDetector = (function() {
+  var map = {}
+
+  // define an InfiniteLoopError class
+  function InfiniteLoopError(msg, type) {
+    Error.call(this ,msg)
+    this.type = 'InfiniteLoopError'
+  }
+  
+  function infiniteLoopDetector(id) {
+    if (id in map) { // Not the first execution, it can be optimized here, the performance is too low
+      if (Date.now() - map[id] > 1000) {
```

Comment:
Hardcoded 1 second? Maybe we can make this adjustable? Or do you think 1 second is fine?